lib/agenda/index.js

```diff
@@ -31,6 +32,14 @@ class Agenda extends Emitter {
     this._sort = config.sort || {nextRunAt: 1, priority: -1};
     this._indices = Object.assign({name: 1}, this._sort, {priority: -1, lockedAt: 1, nextRunAt: 1, disabled: 1});
 
+    if (config.events) {
```

Comment:
This seems somewhat unrelated

Reply:
It's been awhile, and with new changes this may not be needed anymore. But with out this there was no way to guarantee that you could receive an on ready event callback when no index was created. As it was happening synchronously on agenda instance creation.

Reply:
I will review and see if this is still needed or not.

Reply:
Even if this is needed it should be in another PR.

Reply:
Created #626 for this.